pandas/core/dtypes/missing.py

```diff
@@ -63,7 +78,29 @@
 _dtype_str = np.dtype(str)
 
 
-def isna(obj):
+@overload
+def isna(obj: Scalar) -> bool:
+    ...
+
+
+@overload
+def isna(
+    obj: IndexOpsMixin | ExtensionArray | np.ndarray | list,
```

Comment:
IndexOpsMixin overlaps with NDFrameT below.  maybe just Index here?

Reply:
Are there sub-classes of `NDFrame` that inherit from `IndexOpsMixin`?

Used `Index` earlier, but then I needed one mypy ignore for a method in `IndexOpsMixin`.

Reply:
IndexOpsMixin is effectively Union[Index, Series]

Reply:
Thanks! I'm surprised that mypy/pyright didn't complain about that (probably just pick the first matching overload).

Reply:
You can use `IndexOpsMixin` and avoid the `#type: ignore` in `pandas/core/base.py` as follows:
```python
@overload
def notna(obj: Scalar) -> bool:
    ...


@overload
def notna(
    obj: ExtensionArray | np.ndarray | list,
) -> npt.NDArray[np.bool_]:
    ...


@overload
def notna(obj: NDFrameT) -> NDFrameT:
    ...


@overload
def notna(
    obj: IndexOpsMixin,
) -> npt.NDArray[np.bool_]:
    ...


@overload
def notna(obj: object) -> bool | npt.NDArray[np.bool_] | NDFrame:
    ...


def notna(obj: object) -> bool | npt.NDArray[np.bool_] | NDFrame:
```
In this case, `Series`, which is a subclass of `IndexOpsMixin` will match on the `NDFrameT`, but then any other `IndexOpsMixin` (i.e., `Index`) will match after that.